File: frontend/static/frontend/js/components/Home/Calendar.js
import React, { useState, useEffect } from "react";
import { Row, Col, Loading, Text } from "@geist-ui/react";
import { axiosInstance as axios } from "../axiosInstance";

export default function Calendar() {
  let [calendar, setCalendar] = useState({});
  let [loading, setLoading] = useState(true);

  useEffect(() => {
    setLoading(true);
    axios
      .get("calendar")
      .then(res => setCalendar(res.data))
      .catch(err => console.error(err.message))
      .finally(() => setLoading(false));
  }, []);

  return loading ? (
    <Loading size="large" />
  ) : (
    <>
      <Text h1>
        {calendar.monthNames[calendar.monthIndex]} {calendar.year}
      </Text>
      <Row>
        {calendar.dayNames.map(day => (
          <Col>{day}</Col>
        ))}
      </Row>
      {calendar.month.map(week => (
        <Row>
          {week.map(day => (
            <Col>{day === 0 ? null : day}</Col>
          ))}
        </Row>
      ))}
    </>
  );
}

File: backend/calendar.py
import calendar
from datetime import datetime


_now = datetime.now()

MONTH = calendar.monthcalendar(_now.year, _now.month)

for _i, _week in enumerate(MONTH):
    for _j, _day in enumerate(_week):
        MONTH[_i][_j] = dict(
            day=_day,
            fullDate=f'{_now.strftime("%Y-%m")}-{_day:02}',
            today=(_now.day == _day),
        )
